server/metadata.go

```diff
@@ -106,3 +148,22 @@ func (m *Metadata) parseFile(c []byte, fileExt string) (*pbVersion.MetadataVersi
 
 	return meta, nil
 }
+
+func (m *Metadata) parseV2File(c []byte, fileExt string) (*pbVersion.MetadataV2Version, error) {
+	meta := &pbVersion.MetadataV2Version{}
+	switch fileExt {
+	case ".yaml", ".yml":
+		if err := protoyaml.Unmarshal(c, meta); err != nil {
+			return nil, errors.Join(err, errors.New("could not unmarshal yaml"))
+		}
+
+	case ".json":
+		if err := protojson.Unmarshal(c, meta); err != nil {
+			return nil, errors.Join(err, errors.New("could not unmarshal json"))
+		}
+	default:
+		return nil, fmt.Errorf("extension %s not supported", fileExt)
+	}
```

Comment:
this part is duplicated and I think we should enable allowing partial parsing and unknown fields in protojson